File: BOJ/1931_회의실배정/go/Main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {

	var scanner = bufio.NewScanner(os.Stdin)
	scanner.Scan()
	count, _ := strconv.Atoi(scanner.Text())
	var times [][]int
	for i := 0; i < count; i++ {
		scanner.Scan()
		s := strings.Split(scanner.Text(), " ")
		var a, b int
		a, _ = strconv.Atoi(s[0])
		b, _ = strconv.Atoi(s[1])
		var r []int
		r = append(r, a, b)
		times = append(times ,r)
	}

	sort.Slice(times, func(i, j int) bool {
		return times[i][0] < times[j][0]
	})
	sort.Slice(times, func(i, j int) bool {
		return times[i][1] < times[j][1]
	})

	result := 0
	lastEndTime := 0
	for _, v := range times {
		if v[0] >= lastEndTime {
			lastEndTime = v[1]
			result++
		}
	}
	fmt.Print(result)
}

File: BOJ/1931_회의실배정/java/Main.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int count = Integer.parseInt(br.readLine());

        int[][] times = new int[count][2];
        for (int i = 0; i < count; i++) {
            String[] input = br.readLine().split(" ");
            times[i][0] = Integer.parseInt(input[0]);
            times[i][1] = Integer.parseInt(input[1]);
        }

        // 1번째 인덱스로 오름차순 정렬 (같은 값일 경우 0번째 인덱스로 오름차순 정렬)
        Arrays.sort(times, (o1, o2) -> {
            if(o1[1] == o2[1]) {
                return Integer.compare(o1[0], o2[0]);
            } else {
                return Integer.compare(o1[1], o2[1]);
            }
        });

        int result = 0;
        int lastEndTime = 0;
        for (int[] time: times) {
            if (time[0] >= lastEndTime) {
                lastEndTime = time[1];
                result++;
            }
        }

        System.out.print(result);
    }
}
